docs/guidelines.md

```diff
@@ -0,0 +1,130 @@
+# Feedback guidelines
+
+The goal of a language track on Exercism is to give people a way to achieve a
+high level of fluency at a low level of proficiency. We're aiming for fluency
+in the syntax, idioms, and the standard library of the language. You can read
+more about the [goal of exercism here](https://github.com/exercism/docs/blob/master/about/goal-of-exercism.md).
+
+## Definitions
+
+In the following paragraphs, keywords such as **MUST**, **SHOULD**, **COULD**
+are to be interpreted as in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt);
+given that we recognise the following four output states and their restrictions:
+
+- `approve_as_optimal`: **SHOULD** be a best solution, **SHOULD** be without
```

Comment:
The website will drop any comments in this scenario. Does that mean it should be `**MUST** be without comment` or because the website handles, SHOULD is ok?

Reply:
In what scenarios would this not be a best solution? The website can't guard for this, so it should maybe be `**MUST**`?

Reply:
Unfortunately no. It **SHOULD** be, because in fact you and I for example do structural testing. We can not _guarantee_ that all `approve_as_optimal` are _thé_ best solutions there are. We can only add as many assertions as possible that it's likely and probably.

**MUST** here would mean that someone would be able to blindly copy anything that's `approve_as_optimal` and it would _never_ include anything that's less than optimal. The fact that the website can't guard is exactly why it shouldn't be MUST.

> The website will drop any comments in this scenario

You're correct: because the website does not _break_ if you do send it, it's **SHOULD**.

Reply:
> We can not guarantee that all

Which side is that? The analyzer must guarantee that. Or is that the website which is not able to guarantee it?

Reply:
I still think this should be: `**MUST** be a best solution`.

If you can guarantee it or not doesn't matter. We would consider it a bug if it is not a "best solution" and fix it.

**SHOULD** implies that there might be valid reasons to use `approve_as_optimal` although the solution is not optimal. I can't see any.

Definition:
```
3. SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.
```

Reply:
Yes. I've changed my mind now. I agree with both @tehsphinx @iHiD.

Let's make this **MUST** be a best solution. The other terms seem fine.